encrypter.cpp

```diff
@@ -0,0 +1,212 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <ios>
+#include <random>
+#include <sstream>
+#include <utility>
+
+using Byte = char;
+
+//=====put_error=====//
+void put_error(const std::string& error_message)
+{
+  std::cout << error_message;
+  exit(1);
+}
+
+//=====shuffle=====//
+void shuffle(std::vector<Byte>& input)
+{
+  std::default_random_engine gen;
+
+  int i = 0;
+  for (std::vector<Byte>::iterator iter = input.begin(); iter < input.end(); ++i, ++iter)
+  {
+    std::uniform_int_distribution<int> dis(i, input.size() - 1);
+    std::swap(input[i], input[dis(gen)]);
+  } 
+}
+
+//=====replace_extension=====//
+std::string replace_extension(const std::string& filename, const std::string& new_extension)
+{
+  int i = filename.rfind('.');
+  if (i == -1)
+    put_error("A given file doesn't have an extension.\n");
+
+  ++i;
+  if (filename[i] == '.')
+    put_error("Oh, snap! Something went wrong.\n");
+
+  std::string new_name = filename.substr(0, i + 1);
+  std::string::const_iterator iter = new_extension.begin();
+
+  for (int k = 0; iter < new_extension.end(); ++k, ++i, ++iter)
+    new_name[i] = new_extension[k];
+
+  return new_name;
+}
+
+//=====read_file=====//
+std::vector<Byte> read_file(const std::string& filename, const std::string error_message, std::streampos& size)
+{
+  std::ifstream file (filename, std::ios::in|std::ios::binary|std::ios::ate);
+  size = file.tellg();
+  if (!file.is_open())
+    put_error(error_message);
+
+  std::vector<Byte> input(size);
+  file.seekg(0, std::ios::beg);
+  file.read(&input[0], size);
+  file.close();
+
+  if (size == 0)
+    put_error("Given file is empty\n");
+
+  return std::move(input);
+}
+
+//=====write_file=====//
+void write_file(const std::string& filename, std::vector<Byte> output, const std::string error_message, const std::streampos& size)
+{
+  std::ofstream file (filename, std::ios::out|std::ios::binary|std::ios::trunc);
+  if (!file.is_open())
+    put_error(error_message);
+  
+  file.write(&output[0], size);
+}
+
+//=====make_key=====//
+void make_key(const std::string& filename)
+{
+  std::streampos size;
+  std::vector<Byte> input = read_file(filename, "Couldn't access file-base for a key.\n", size);
```

Comment:
It would be more readable and shorter to return size and take vector& as argument.